### PR TITLE
Check translation progress during final release

### PR DESCRIPTION
**Pull request by loremattei**
This PR adds a step to the `finalize_release` lane to check the current state of the translation.
If some of the languages which are expected to be fully translated at release time are below 100%, the lane outputs the current state of the translation and asks confirmation to the user before continuing.

```
### Example:

[15:53:43]: ----------------------------------------
[15:53:43]: --- Step: check_translation_progress ---
[15:53:43]: ----------------------------------------
[15:53:43]: Checking translations status...
[15:53:44]: > Getting translation status for ar
[15:53:44]: Language ar is 99% translated.
[15:53:44]: > Getting translation status for de
[15:53:44]: Language de is 97% translated.
[15:53:44]: > Getting translation status for es
[15:53:44]: Language es is 100% translated.
[15:53:44]: > Getting translation status for fr
[15:53:44]: Language fr is 96% translated.
[15:53:44]: > Getting translation status for he
[15:53:44]: Language he is 97% translated.
[15:53:44]: > Getting translation status for id
[15:53:44]: Language id is 96% translated.
[15:53:44]: > Getting translation status for it
[15:53:44]: Language it is 97% translated.
[15:53:44]: > Getting translation status for ja
[15:53:44]: Language ja is 99% translated.
[15:53:44]: > Getting translation status for ko
[15:53:44]: Language ko is 97% translated.
[15:53:44]: > Getting translation status for nl
[15:53:44]: Language nl is 100% translated.
[15:53:44]: > Getting translation status for pt-br
[15:53:44]: Language pt-br is 98% translated.
[15:53:44]: > Getting translation status for ru
[15:53:44]: Language ru is 100% translated.
[15:53:44]: > Getting translation status for sv
[15:53:44]: Language sv is 100% translated.
[15:53:44]: > Getting translation status for tr
[15:53:44]: Language tr is 99% translated.
[15:53:44]: > Getting translation status for zh-cn
[15:53:44]: Language zh-cn is 100% translated.
[15:53:44]: > Getting translation status for zh-tw
[15:53:44]: Language zh-tw is 97% translated.
[15:53:44]: The translations for the following languages are below the 100% threshold:
 - ar is at 99%.
 - de is at 97%.
 - fr is at 96%.
 - he is at 97%.
 - id is at 96%.
 - it is at 97%.
 - ja is at 99%.
 - ko is at 97%.
 - pt-br is at 98%.
 - tr is at 99%.
 - zh-tw is at 97%.
Do you want to continue? (y/n)
```

## Regression Notes
1. Potential unintended areas of impact
This PR updates `release-toolkit` to version `1.3.1`, so it brings in some other changes which impact the Android release lanes. I'm explicitly adding @AliSoftware as a reviewer of this PR because of this. 

BTW: I've noticed that the `hotfix` check has been removed from `finalize_release`, so I've updated my changes accordingly. I'm mentioning it to double check that it was intentional.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
-

3. What automated tests I added (or what prevented me from doing so)
-

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AliSoftware** [comment]
@loremattei I see the similar change was merged for WPiOS, so just wondering if you forgot about undrafting this one and it's ready for review… or if you indeed still have work to do on it 🙂 

I see you set this PR's milestone to 17.7, but in case you think it will be ready by 17.6, feel free to retarget this PR to `release/17.6` (if it can make it to this release now that the branch is cut, that is); if not, it's ok, we'll just land it in 17.7 as the milestone suggests 🙂

**Comment by loremattei** [comment]
@AliSoftware thanks for checking. I opened similar PRs in every repository, but before merging them I want to monitor the tool on WPiOS for one full cycle, because the feature relies on the hack of parsing a HTML page given that GlotPress doesn't have an API.

**Comment by AliSoftware** [comment]
Make sense, thanks for the update 👍

**Comment by AliSoftware** [comment]
Moving milestone to 17.8 as 17.7 ❄️  is happening today

**Comment by AliSoftware** [comment]
> BTW: I've noticed that the hotfix check has been removed from finalize_release, so I've updated my changes accordingly. I'm mentioning it to double check that it was intentional.

It was changed into a guard statement at the top of the lane, because we have a separate `finalize_hotfix_release` for those cases. See [this line](https://github.com/wordpress-mobile/WordPress-Android/pull/14828/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688R364)

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile:380]: 1. I think it would be wise to also check the translations of the Release Notes GP project too, in addition to the one about in-app strings? wdyt?
2. If so, the URL is different between WP and JP for Release Notes GP projects, so we'd want to use the right one depending on the `app:` parameter used. There are already constants in `APP_SPECIFIC_VALUES` for each of those.

**Review by AliSoftware — COMMENTED** [review]
FYI I'm waiting for https://github.com/wordpress-mobile/WordPress-Android/pull/15011/ to land first because there might be conflicts between those too – as the release-toolkit was already updated to 1.3.1 in the `release/17.7` since that's where I used the toolkit last.

So I prefer to land https://github.com/wordpress-mobile/WordPress-Android/pull/15011/ first (since that one will be a blocker for Monday's code freeze if not merged), and fix the conflict on that one, than the other way around (which would otherwise require me to open a separate PR for the release->develop to solve the conflict without merging back develop into release).

**Comment by AliSoftware** [comment]
Moving this to 17.9 since it didn't make it to 17.8 code freeze, and is pending addressing feedback and conflict.

File: fastlane/Fastfile
Comment (loremattei) [reply fastlane/Fastfile:380]: Sg. I didn't add it because it's something the release manager usually wants to check constantly during the release cycle, but if you think it can be useful, I have no strong opinions against it.

Comment (AliSoftware) [reply fastlane/Fastfile:380]: Yeah usually I do check those especially the morning before the submission (in preparation for the submission I do the Friday afternoon) so I can ping #i18n if some are missing. But it couldn't hurt to have that additional, automated check in case the RM forgot, better twice than none 🙂

File: fastlane/Fastfile
Comment (loremattei) [inline fastlane/Fastfile:373]: I've seen this is done in other lanes, so I added it for consistency

Comment (AliSoftware) [reply fastlane/Fastfile:373]: 👍  This will be refactored soon (once I get time to undo/redo the old approach that was added around `app:` parameter and instead make the various lanes always act on both WP and JP at once), but it's good to have in the meantime.

**Review by AliSoftware — APPROVED** [review]
LGTM now 👍 

Only change that could be worth doing before merging is to consider changing the PR's target to `release/17.8` so that it would already be used for the current 17.8 release from this cycle.

**Comment by AliSoftware** [comment]
As for the CI test failure, see p1626093894408000-slack-C011BKNU1V5

**Comment by loremattei** [comment]
@AliSoftware I tried to move this PR to `release/17.8`, but it's not great. 
New PRs have been merged into `develop` after the code freeze, so I had to rebase my changes on top of `release/17.8` in order to be able to switch branch without adding unwanted changes to the release branch. 
But, after the rebase, the diff shown in GitHub is quite confusing, because as usual in these cases, GitHub gets confused and shows more changes than the ones in the PR (like: it shows as changes all the solved merge conflicts, even if the version in the target branch was accepted, because it has a different commit hash).
 
I don't like it, because it makes it difficult to understand what happened if we'll go back to it in the future. So, I moved things back to `develop`. Since the changes in this PR are useful, but not critical, I'd prefer to keep things clean. 
Sgty?

**Comment by AliSoftware** [comment]
> the diff shown in GitHub is quite confusing, because as usual in these cases, GitHub gets confused and shows more changes than the ones in the PR

Mmmh yeah. My current recurring problem with GitHub 😒  (especially while working on tooling PR and on multiple repos… each evolving at their pace while the PRs like that that we mirror on each repo are waiting to be merged…)

Let's merge it to `develop` then, not critical as you say, so SGTM 👍

**Comment by AliSoftware** [comment]
https://github.com/wordpress-mobile/WordPress-Android/pull/15019 is pending to fix the test failures. Once it's merged into `release/17.8` and then we merge `release/17.8` into `develop` via https://github.com/wordpress-mobile/WordPress-Android/pull/15018/, `develop` should be fixed and we could sync this PR to get the CI go green there too.

(Alternatively, we could just merge this PR directly considering that the test failures on CI are unrelated so it's ok to land regardless)

**Comment by loremattei** [comment]
@AliSoftware I've merged the latest `develop` (which contains the fix you mentioned) back into this PR and now it's all green as expected. 
I'm merging this.

**Pull request merged**
